zazu/dev/commands.py

```diff
@@ -230,19 +233,33 @@ def get_pulls_for_branch(branch):
 
 @dev.command()
 @click.pass_context
-def review(ctx):
+@click.option('--base', default='develop', help='The base branch to target')
+@click.option('--head', default='', help='The head branch (defaults to current branch)')
+def review(ctx, base, head):
     """Create or display pull request"""
-    import urllib
-    encoded_branch = urllib.quote_plus(ctx.obj.repo.active_branch.name)
+    # TODO(nwiles): Refactor this into a plugin.
     url = ctx.obj.repo.remotes.origin.url
-    start = 'github.com'
-    if start in url:
-        base_url = url[url.find(start):].replace('.git', '').replace(':', '/')
-        url = 'https://{}/compare/{}?expand=1'.format(base_url, encoded_branch)
-        click.echo('Opening "{}"'.format(url))
-        webbrowser.open_new(url)
-        # TODO: add link to ticket in the PR, zazu logo
-        # <img src="http://vignette1.wikia.nocookie.net/disney/images/c/ca/Zazu01cf.png" alt="Zazu" width=50"/>
+    if 'github.com' in url:
+        if not head:
+            head = ctx.obj.repo.active_branch.name
+        gh = zazu.github_helper.make_gh()
+        owner, repo_name = zazu.github_helper.parse_github_url(url)
+        repo = gh.get_user(owner).get_repo(repo_name)
+        existing_pulls = repo.get_pulls(state='open', base=base, head=head)
+        try:
+            pr = existing_pulls[0]
+        except IndexError:
+            click.echo('No existing review found, creating one...')
```

Comment:
So existing_pulls contains all PRs that match the filter criteria of the query (note that the head and base are part of the query). This allows the filtering to be done server side. There was still an issue though that has now been fixed. The head must be specified in the form *org:branch*. If the org was missing github disregarded that filter entirely, giving you all PRs targeting develop.